### PR TITLE
Add exhaustive search creation UI

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/55707
Partially closes https://github.com/sourcegraph/sourcegraph/issues/55973

[Figma desings](https://www.figma.com/file/H4ezuXmEvVeoC6UaiNSbEa/Exhaustive-search-UI?type=design&node-id=1568-10052&mode=design&t=3su0QK4wsCCTC082-0)

| New UI with no problems in query | With validation problems |
| -----  | ----- | 
| Screenshot 2023-08-25 at 00 57 36 | Screenshot 2023-08-25 at 00 57 51 |

## Tech notes 
In this PR we add an initial creation form UI in skipped results popover and also add some query validation algoright (https://github.com/sourcegraph/sourcegraph/issues/55973), but this validation isn't comprehensive since we probably want to check the query in a more sophisticated way (like analyse OR operand and check filter in operands like @stefanhengl suggested in this [comment](https://github.com/sourcegraph/sourcegraph/issues/55973#issuecomment-1688429040) and not just check operator existence in the query) 

## Design review notes 
- The experimental badge has a different colour/font capital compared to what we have in design, @rrhyne let me know if it's a misalignment in wildcard implementation, our storybook [demo](https://storybook.sgdev.org/?path=/story/wildcard-productstatusbadge--badges)
-  In Figma, we have just block with an alert warning border but not an Alert warning border. I used the Alert component here since we have [them in the wildcard](https://storybook.sgdev.org/?path=/story/wildcard-alert--alerts)

## Test plan
- Turn on `exhaustiveSearch:true` in your `experimentalFeatures` in your settings editor
- Go to search and try to search for something that may produce more results that we can show on the search result page 
- Check the skipped results button. There should be an exhaustive search (search job) creation UI

### CODE REVIEW

File: schema/settings.schema.json [hunk 1]
```diff
@@ -17,7 +17,19 @@
...
+            "old-strict-list",
+            "single-search-query",
+            "search-query-or-strict-list"
+          ],
+          "!go": {
+            "pointer": true
+          }
+        },
+        "exhaustiveSearch": {
+          "description": "Enables exhaustive search feature and its UI",
```
Comment (stefanhengl) [inline schema/settings.schema.json:30]: Shall we call this "search jobs" from now on?

**Review by stefanhengl — APPROVED** [review]
Approving to unblock. The changes are behind a feature flag so it should be ok :-)  Based on the workflow I believe it might make sense to move the validation to the API but I think for now it's great to have it working in the client already.

File: schema/settings.schema.json (hunk 1 above)
Comment (vovakulikov) [reply schema/settings.schema.json]: True, just updated, thanks for the catch

**Comment by rrhyne** [comment]
Vova, I'm seeing some misalignment between the screenshots and the Figma. Things like the badge state, the button color, etc. 

Can you try to align closely to the designs?

**Comment by vovakulikov** [comment]
@rrhyne i listed some misalignments in this PR description in design review section, I think there's some problem in Figma and wildcard components, I can fix this but these problems are common in other places in the app, I would prefer to avoid changing common UI in badges and alerts UI. But if you think that Figma components are correct we can fix it of course, thanks

**Comment by vovakulikov** [comment]
@rrhyne I fixed button size, variants and copy. Thanks for the catch. But the alert UI and experiment badge come from the shared UI, as I mentioned above and in the PR description originally. I will merge this one as is, and let's check Figma UI components and adjust it in the UI in further PR (fix should be 5 min change) 😉

**Comment by rrhyne** [comment]
Thanks Vova. The experimental badge you used is correct, I supplied an incorrect version.

**Pull request merged**